Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -388,33 +388,61 @@ public class ProgramBuilder {
     /// Append a splice from another program.
     public func splice(from program: Program, at index: Int) {
         var idx = index
-        
+
+        // Get types of the other program
+        var ai = AbstractInterpreter(for: self.fuzzer)
+
+        for instr in program {
+            ai.execute(instr)
```

Comment:
I think it would be easier to just use the AI unconditionally here and then do something like:
```
let type = program.runtimeTypes[input] ?? ai.type(of: input)
```
Currently the `runtimeTypes` map can be accessed by a function which returns the type or `.unknown` in which case I think the AI might provide better type info, which is why I am accessing it here directly to get an optional value.

Then we could also do something like:
```
let replacementVar = randVar(ofConservativeType: type) ?? randVar(ofConservativeType: type.generalize())
if let hostVar = replacementVar {
    // do rewire
}
```
Or we could try to aggregate all the type info in the AI like you said and use the AI's interface for type queries? 
What do you think? :)

Reply:
Yeah I guess we can leave it like this until we have a proper type information API that incorporates both interpreter-inferred and runtime-collected type information :)